examples/editor/index.js

```diff
@@ -365,3 +369,101 @@ document.getElementById('fullscreen').onclick = () => {
     document.querySelector('.map-info').style.display = 'none';
     map.resize();
 };
+
+const $exportMapButton = document.getElementById('export-map-button');
+const $copyHTMLButton = document.getElementById('copy-html-button');
+const $mapTextarea = document.getElementById('map-textarea');
+
+$exportMapButton.addEventListener('click', () => {
+    const config = getJSONConfig();
+    $mapTextarea.value = generateSnippet(config);
+});
+
+$copyHTMLButton.addEventListener('click', () => {
+    $mapTextarea.select();
+    document.execCommand('copy');
+});
+
+
+/**
+
+
+"a":"sf_stclines",
+"b":"",
+"c":"dmanzanares",
+"d":"carto.com",
+"e":"color: ramp($st_type, prism) \nwidth: 1.5",
+"f":{
+    "lng":-122.44408486861192,
+    "lat":37.773706736149705
+    },
+ "g":11.664310802866805
+}
+
+* @param {*} config 
+ * 
+ */
+function generateSnippet(config) {
+    const dataset = config.a;
+    const apiKey = config.b || 'default_public';
+    const username = config.c;
+    const serverURL = config.d || 'https://{user}.carto.com';
+    const vizSpec = config.e || '';
+    const center = config.f || { lat: 0, lng: 0 };
+    const zoom = config.g || 10;
+    // const basemap = config.h || 'dark-matter';
```

Comment:
?

Reply:
2 options here:

1. Add the real name of the basemap and replace the string
2. Add an "alias" and use a switch statement to get the real map URL

(I prefer the second one)